geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientCQClusterRestartDUnitTest.java

```diff
@@ -0,0 +1,315 @@
+    server.bounceForcibly();
+    locator.invoke(this::stopLocator);
+  }
+
+  private int startLocator(File locatorLog) throws IOException {
+    InetAddress bindAddress = InetAddress.getByName(hostName);
+    Locator locator =
+        Locator.startLocatorAndDS(locatorPort, locatorLog, bindAddress, new Properties());
+    return locator.getPort();
+  }
+
+  private void createCacheServerAndDiskRegion(File logFile) throws Exception {
+    Properties systemProperties = new Properties();
+    systemProperties.setProperty("gemfire.jg-bind-port", Integer.toString(uniquePort));
+    cacheRule.createCache(createServerConfig(logFile), systemProperties);
+
+    createDiskRegionIfNotExist();
+
+    CacheServer server = cacheRule.getCache().addCacheServer();
+    server.setPort(0);
+    server.start();
+  }
+
+  private Properties createServerConfig(File logFile) {
+    Properties config = new Properties();
+    config.setProperty(LOCATORS, hostName + "[" + locatorPort + "]");
+    config.setProperty(LOG_FILE, logFile.getAbsolutePath());
+    return config;
+  }
+
+  private void closeCache() {
+    cacheRule.closeAndNullCache();
+  }
+
+  private void stopLocator() {
+    Locator.getLocator().stop();
+  }
+
+  private void createDiskRegionIfNotExist() throws IOException {
+    if (cacheRule.getCache().getRegion(regionName) != null) {
+      return;
+    }
+    DiskStoreAttributes diskStoreAttributes = new DiskStoreAttributes();
+    DiskStoreFactory diskStoreFactory =
+        cacheRule.getCache().createDiskStoreFactory(diskStoreAttributes);
+    diskStoreFactory.setDiskDirs(new File[] {createOrGetDir()});
+    diskStoreFactory.create(getDiskStoreName());
+
+    RegionFactory regionFactory = cacheRule.getCache().createRegionFactory(REPLICATE_PERSISTENT);
+    regionFactory.setDiskStoreName(getDiskStoreName());
+    regionFactory.create(regionName);
+  }
+
+  private File createOrGetDir() throws IOException {
+    File dir = new File(temporaryFolder.getRoot(), getDiskStoreName());
+    if (!dir.exists()) {
+      dir = temporaryFolder.newFolder(getDiskStoreName());
+    }
+    return dir;
+  }
+
+  private String getDiskStoreName() {
+    return getClass().getSimpleName() + VM.getVMId();
+  }
+
+  private void callFunctions() throws Exception {
+    InternalRegion region = (InternalRegion) clientCacheRule.getClientCache().getRegion(regionName);
+    Pool pool = region.getServerProxy().getPool();
+
+    List<Future<Void>> futures = new ArrayList<>();
+    for (int i = 0; i < numOfInvocations; i++) {
+      futures.add(executorServiceRule.submit(() -> invokeFunction(pool)));
+    }
+    for (Future<Void> future : futures) {
+      future.get(getTimeout().toMillis(), MILLISECONDS);
+    }
+  }
+
+  private void invokeFunction(Pool pool) {
+    @SuppressWarnings("unchecked")
+    Execution execution = FunctionService.onServer(pool).setArguments(regionName);
+    ResultCollector resultCollector = execution.execute(new TestFunction());
```

Comment:
Warnings here can be fixed by using `Execution<String, Void, Void>` and `ResultCollector<Void, Void>`